### ISSUE TITLE
Make tests more robust with environment variable checks

### ISSUE THREAD

**Issue by shubham1172** [dapr/cli#916]
## Background

Environment variables `DAPR_RUNTIME_VERSION` and `DAPR_DASHBOARD_VERSION` are used by e2e tests.

https://github.com/dapr/cli/blob/68c2ab84d48892a0deb42e8cb4c3b1fbc356927c/tests/e2e/kubernetes/kubernetes_test.go#L26-L29

https://github.com/dapr/cli/blob/050eb553f2304aca08cdcbae86f44a97552fa121/tests/e2e/standalone/standalone_test.go#L43-L47

Locally, when someone runs `make test-e2e-sh` without these environment variables set, it fails with the following error:
![image](https://user-images.githubusercontent.com/22132944/157035913-9c22d5d5-8688-487e-925d-e6a4883e7e88.png)

This does not give any idea on what the actual problem is.

<details>
<summary>CLI logs | head :</summary>

```bash
dapr ➜ /go/src/github.com/dapr/cli (master) $ make test-e2e-sh
# The desire here is to download this test dependency without polluting go.mod
# In golang >=1.16 there is a new way to do this with `go install gotest.tools/gotestsum@latest`
# But this doesn't work with <=1.15, so we do it the old way for now 
# (see: https://golang.org/ref/mod#go-install)
GO111MODULE=off go get gotest.tools/gotestsum
gotestsum --jsonfile test_output.json --format standard-verbose -- -count=1 -tags=e2e ./tests/e2e/standalone/...
=== RUN   TestStandaloneInstall
=== RUN   TestStandaloneInstall/test_install
    standalone_test.go:190: {"time":"2022-03-07T12:36:21.3888015Z","status":"pending","msg":"Making the jump to hyperspace..."}
        {"time":"2022-03-07T12:36:21.3943455Z","status":"info","msg":"Installing runtime version "}
        {"time":"2022-03-07T12:36:22.1383481Z","status":"pending","msg":"Downloading binaries and setting up components..."}
        {"time":"2022-03-07T12:36:22.2067208Z","status":"failure","msg":"Downloading binaries and setting up components..."}
        {"time":"2022-03-07T12:36:22.2067746Z","status":"failure","msg":"docker: invalid reference format.\nSee 'docker run --help'.\n"}
        
    standalone_test.go:191: 
                Error Trace:    standalone_test.go:191
                Error:          Received unexpected error:
                                exit status 1
                Test:           TestStandaloneInstall/test_install
                Messages:       init failed
=== RUN   TestStandaloneInstall/test_install_from_custom_registry
    standalone_test.go:199: Custom image registry is not set, skipping the test for now..
=== RUN   TestStandaloneInstall/test_run
=== RUN   TestStandaloneInstall/test_run/Normal_exit,_socket:_
    standalone_test.go:386: ℹ️  Starting Dapr with id Hyenaberyl-Pixie. HTTP Port: 34507. gRPC Port: 44421
        ❌  fork/exec /home/dapr/.dapr/bin/daprd: no such file or directory
        
    standalone_test.go:387: 
                Error Trace:    standalone_test.go:387
                Error:          Received unexpected error:
                                exit status 1
                Test:           TestStandaloneInstall/test_run/Normal_exit,_socket:_
                Messages:       run failed
```
</details>

## Fix

Add checks for environment variables in the tests initialization and fail with a proper error message with they are missing.

**Comment by hueifeng** [comment]
@mukundansundar Can this issue be closed? see: https://github.com/dapr/cli/pull/945

**Comment by mukundansundar** [comment]
> @mukundansundar Can this issue be closed? see: #945

closing as part of previous milestone work.
